src/PageCharacter/CharacterDisplay/Tabs/TabOptimize/index.tsx

```diff
@@ -454,12 +454,29 @@ function BuildList({ buildsArts, characterKey, data, compareData, disabled }: {
       build={build}
       oldData={data}
     >
-      <BuildDisplayItem index={index} compareBuild={compareData} disabled={disabled} />
+      <BuildItemWrapper index={index} build={build} compareData={compareData} disabled={disabled} />
     </DataContextWrapper>
     )}
   </Suspense>, [buildsArts, characterKey, data, compareData, disabled])
   return list
 }
+function BuildItemWrapper({ index, build, compareData, disabled }: {
+  index: number
+  build: ICachedArtifact[],
+  compareData: boolean,
+  disabled: boolean,
+}) {
+  const location = useLocation()
+  const navigate = useNavigate()
+  const toTC = useCallback(() => {
+    const paths = location.pathname.split("/")
+    paths.pop()
+    navigate(`${paths.join("/")}/theorycraft`, { state: { build } })
+  }, [navigate])
+
+  return <BuildDisplayItem index={index} compareBuild={compareData} disabled={disabled}
+    extraButtonsLeft={<Button color="info" startIcon={<Science />} onClick={toTC}>Theorycraft</Button>} />
```

Comment:
Think this needs a `size="small"` to prevent the elements from being different heights 
<img width="998" alt="image" src="https://user-images.githubusercontent.com/36019388/192127257-aec340c3-b37e-402a-8063-f6024a90525a.png">